### ISSUE TITLE
NullReferenceException in SequencePointBuilder.AddToSequencePoint

### ISSUE THREAD

**Issue by yyjdelete** [icsharpcode/ILSpy#2062]
Similar to #1851, but happened with fixed buffers.

### Steps to reproduce
1. Save pdb or `IL with C#` with the below code
```cs
    public unsafe struct TestStruct
    {
        fixed int p2[2];
        public void ZZZ()
        {
            Console.WriteLine(p2[1]);
        }
    }
```

### Error message shown

```cs
System.NullReferenceException: 
   at ICSharpCode.Decompiler.CSharp.SequencePointBuilder.AddToSequencePoint(ILInstruction inst)
   at ICSharpCode.Decompiler.CSharp.SequencePointBuilder.AddToSequencePoint(AstNode node)
   at ICSharpCode.Decompiler.CSharp.SequencePointBuilder.VisitChildren(AstNode node)
   at ICSharpCode.Decompiler.CSharp.Syntax.DepthFirstAstVisitor.VisitPrimitiveExpression(PrimitiveExpression primitiveExpression)
   at ICSharpCode.Decompiler.CSharp.Syntax.PrimitiveExpression.AcceptVisitor(IAstVisitor visitor)
   at ICSharpCode.Decompiler.CSharp.Syntax.DepthFirstAstVisitor.VisitChildren(AstNode node)
   at ICSharpCode.Decompiler.CSharp.SequencePointBuilder.VisitChildren(AstNode node)
   at ICSharpCode.Decompiler.CSharp.Syntax.DepthFirstAstVisitor.VisitIndexerExpression(IndexerExpression indexerExpression)
   at ICSharpCode.Decompiler.CSharp.Syntax.IndexerExpression.AcceptVisitor(IAstVisitor visitor)
   at ICSharpCode.Decompiler.CSharp.Syntax.DepthFirstAstVisitor.VisitChildren(AstNode node)
   at ICSharpCode.Decompiler.CSharp.SequencePointBuilder.VisitChildren(AstNode node)
   at ICSharpCode.Decompiler.CSharp.Syntax.DepthFirstAstVisitor.VisitInvocationExpression(InvocationExpression invocationExpression)
   at ICSharpCode.Decompiler.CSharp.Syntax.InvocationExpression.AcceptVisitor(IAstVisitor visitor)
   at ICSharpCode.Decompiler.CSharp.Syntax.DepthFirstAstVisitor.VisitChildren(AstNode node)
   at ICSharpCode.Decompiler.CSharp.SequencePointBuilder.VisitChildren(AstNode node)
   at ICSharpCode.Decompiler.CSharp.Syntax.DepthFirstAstVisitor.VisitExpressionStatement(ExpressionStatement expressionStatement)
   at ICSharpCode.Decompiler.CSharp.Syntax.ExpressionStatement.AcceptVisitor(IAstVisitor visitor)
   at ICSharpCode.Decompiler.CSharp.SequencePointBuilder.VisitAsSequencePoint(AstNode node)
   at ICSharpCode.Decompiler.CSharp.SequencePointBuilder.VisitBlockStatement(BlockStatement blockStatement)
   at ICSharpCode.Decompiler.CSharp.Syntax.BlockStatement.AcceptVisitor(IAstVisitor visitor)
   at ICSharpCode.Decompiler.CSharp.Syntax.DepthFirstAstVisitor.VisitChildren(AstNode node)
   at ICSharpCode.Decompiler.CSharp.SequencePointBuilder.VisitChildren(AstNode node)
   at ICSharpCode.Decompiler.CSharp.Syntax.DepthFirstAstVisitor.VisitMethodDeclaration(MethodDeclaration methodDeclaration)
   at ICSharpCode.Decompiler.CSharp.Syntax.MethodDeclaration.AcceptVisitor(IAstVisitor visitor)
   at ICSharpCode.Decompiler.CSharp.Syntax.DepthFirstAstVisitor.VisitChildren(AstNode node)
   at ICSharpCode.Decompiler.CSharp.SequencePointBuilder.VisitChildren(AstNode node)
   at ICSharpCode.Decompiler.CSharp.Syntax.DepthFirstAstVisitor.VisitSyntaxTree(SyntaxTree syntaxTree)
   at ICSharpCode.Decompiler.CSharp.Syntax.SyntaxTree.AcceptVisitor(IAstVisitor visitor)
   at ICSharpCode.Decompiler.CSharp.CSharpDecompiler.CreateSequencePoints(SyntaxTree syntaxTree)
   at ICSharpCode.ILSpy.CSharpILMixedLanguage.MixedMethodBodyDisassembler.Disassemble(PEFile module, MethodDefinitionHandle handle)
   at ICSharpCode.Decompiler.Disassembler.ReflectionDisassembler.DisassembleMethodBlock(PEFile module, MethodDefinitionHandle handle, GenericContext genericContext)
   at ICSharpCode.Decompiler.Disassembler.ReflectionDisassembler.DisassembleMethod(PEFile module, MethodDefinitionHandle handle)
   at ICSharpCode.ILSpy.ILLanguage.DecompileMethod(IMethod method, ITextOutput output, DecompilationOptions options)
   at ICSharpCode.ILSpy.TreeNodes.MethodTreeNode.Decompile(Language language, ITextOutput output, DecompilationOptions options)
   at ICSharpCode.ILSpy.TextView.DecompilerTextView.DecompileNodes(DecompilationContext context, ITextOutput textOutput)
   at ICSharpCode.ILSpy.TextView.DecompilerTextView.<>c__DisplayClass48_0.<DecompileAsync>b__0()
```

### Details
* Product in use: ILSpy 
* Version in use: 6.0.0.5839

**Comment by siegfriedpammer** [comment]
@dgrunwald Do you think we could find a way to avoid these crashes? i.e. find a way to detect the current ILFunction even if the current instruction is detached from the ILAst? We have some places (e.g., `PointerArithmeticOffset.Detect`) where new instructions are generated that are not connected to any ILFunction, but still end up in the annotations of the C# AST, causing these bugs.

**Comment by siegfriedpammer** [comment]
Should be fixed by 69cad7527edd066aab107e271f31201725fe6b45.
